src/components/order-service/order-service.controller.ts

```diff
@@ -312,6 +313,34 @@ export class OrderServiceController implements angular.IController {
     this.ctrl.parameterSchema = schema;
   }
 
+  private updateParameterForm() {
+    let formDefinition: any = _.get(this, 'ctrl.selectedPlan.externalMetadata.schemas.service_instance.create.form_definition');
+    this.removeUnsupportedParameterFormItems(formDefinition);
+    this.ctrl.parameterForm = formDefinition || ['*'];
+  }
+
+  private removeUnsupportedParameterFormItems(formItems: any) {
+    const allowedKeys = {
+      'key': true,
+      'type': true,
```

Comment:
I'd like to lock down the possible values for type as well. I think this would be a good set:

- fieldset
- text
- textarea
- password
- checkbox
- select

https://github.com/json-schema-form/angular-schema-form/blob/development/docs/index.md#form-types

We might need to move away from angular-schema-form someday, and this list seems manageable if we ever need to implement it ourselves.